DisguiseUnityRenderStream/Runtime/PipelineAbstraction/DepthCopy.cs

```diff
@@ -0,0 +1,193 @@
+            {
+                m_Raw.Create(shaderPath, shaderPass);
+                m_Eye.Create(shaderPath, shaderPass);
+                m_linear01.Create(shaderPath, shaderPass);
+                
+                m_Raw.SetKeyword("DEPTH_COPY_RAW", true);
+                m_Eye.SetKeyword("DEPTH_COPY_EYE", true);
+                m_linear01.SetKeyword("DEPTH_COPY_LINEAR01", true);
+            }
+
+            public void Dispose()
+            {
+                m_Raw.Dispose();
+                m_Eye.Dispose();
+                m_linear01.Dispose();
+            }
+        }
+        
+        struct ShaderResources : IDisposable
+        {
+            public Shader m_shader;
+            public Material m_material;
+            public int m_pass;
+
+            public bool IsLoaded => m_shader != null && m_material != null && m_pass >= 0;
+
+            public void SetKeyword(string keyword, bool value)
+            {
+                if (IsLoaded)
+                {
+                    if (value)
+                        m_material.EnableKeyword(keyword);
+                    else
+                        m_material.DisableKeyword(keyword);
+                }
+            }
+
+            public void Create(string shaderPath, string shaderPass)
+            {
+                m_shader = Shader.Find(shaderPath);
+                if (m_shader != null)
+                {
+                    m_material = CoreUtils.CreateEngineMaterial(m_shader);
+                    m_pass = m_material.FindPass(shaderPass);
+                }
+            }
+
+            public void Dispose()
+            {
+                CoreUtils.Destroy(m_material);
+            }
+        }
+
+        public Mode mode
+        {
+            get => m_mode;
+            set
+            {
+                m_mode = value;
+                
+                switch (m_mode)
+                {
+                    case Mode.Raw:
+                        m_shaderResources = s_shaderVariantResources.m_Raw;
+                        break;
+                    case Mode.Eye:
+                        m_shaderResources = s_shaderVariantResources.m_Eye;
+                        break;
+                    case Mode.Linear01:
+                        m_shaderResources = s_shaderVariantResources.m_linear01;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+        }
+        
+        const string k_profilerTag = "Disguise Depth Copy";
+        const string k_ShaderPass = "Depth Copy";
+
+        static ShaderVariantResources s_shaderVariantResources;
+
+        Mode m_mode = Mode.Linear01;
+        ShaderResources m_shaderResources;
+
+        public DepthCopy()
+        {
+            if (!s_shaderVariantResources.IsLoaded)
+            {
+                s_shaderVariantResources.Create(k_ShaderName, k_ShaderPass);
```

Comment:
Consider putting static initialization in a static constructor. It's guaranteed to run only once just before the class is used for the first time.
```csharp
static DepthCopy()
{
    s_shaderVariantResources.Create(k_ShaderName, k_ShaderPass);
    Assert.IsTrue(s_shaderVariantResources.IsLoaded, "Couldn't load the shader resources for DepthCopy");
}
```